tools/c7n_azure/c7n_azure/resources/sqlserver.py

```diff
@@ -23,3 +33,89 @@ class resource_type(ArmResourceManager.resource_type):
         service = 'azure.mgmt.sql'
         client = 'SqlManagementClient'
         enum_spec = ('servers', 'list', None)
+
+
+@SqlServer.filter_registry.register('firewall-rules')
+class SqlServerFirewallRulesFilter(Filter):
+    """Filters SQL servers by the firewall rules
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: servers-with-firewall
+                resource: azure.sqlserver
+                filters:
+                  - type: firewall-rules
+                    include:
+                    - '131.107.160.2-131.107.160.3'
+                    - 10.20.20.0/24
+    """
+
+    schema = type_schema(
+        'firewall-rules',
+        **{
+            'include': {'type': 'array', 'items': {'type': 'string'}},
+            'equal': {'type': 'array', 'items': {'type': 'string'}}
+        })
+
+    def __init__(self, data, manager=None):
+        super(SqlServerFirewallRulesFilter, self).__init__(data, manager)
+        self.policy_include = None
+        self.policy_equal = None
+        self.client = self.manager.get_client()
```

Comment:
Can you please try to run `custodian validate`? IIRC, we don't initialize client in the __init__ because it doesn't work with this mode.

Reply:
moved down to process method